### PR TITLE
WIP: Azure Private

### PR DESCRIPTION
**Pull request by Amoghrd**

### CODE REVIEW

File: ci-operator/step-registry/azure/bastion/pre/azure-bastion-pre-commands.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+EOF
```
Comment (liangxia) [inline ci-operator/step-registry/azure/bastion/pre/azure-bastion-pre-commands.sh:7]: The test show `INFO[2022-01-12T20:51:38Z] /bin/bash: line 8: EOF: command not found`

File: ci-operator/step-registry/azure/bastion/pre/azure-bastion-pre-commands.sh [hunk 2]
```diff
@@ -0,0 +1,174 @@
+#!/bin/bash
```
Comment (jianlinliu) [inline ci-operator/step-registry/azure/bastion/pre/azure-bastion-pre-commands.sh:1]: For the script path, I suggest `ci-operator/step-registry/azure/provision/bastion/azure-bastion-commands.sh`, the similar folder structure and files like what aws is doing in https://github.com/openshift/release/pull/22260/files.

File: ci-operator/step-registry/azure/bastion/pre/azure-bastion-pre-ref.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    namespace: ocp
+    name: "4.10"
+    tag: upi-installer
+  grace_period: 10m
+  commands: azure-bastion-pre-commands.sh
+  resources:
+    requests:
+      cpu: 10m
+      memory: 100Mi
+  env:
```
Comment (jianlinliu) [inline ci-operator/step-registry/azure/bastion/pre/azure-bastion-pre-ref.yaml:13]: The bastion ref is only used for provision/deprovision bastion host, should not touched with any install-config.yaml, so I think these parameters are not needed here. I also gave the same comments in gcp cases.

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/private/cucushift-installer-rehearse-azure-ipi-private-workflow.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    - chain: ipi-conf-azure-sharednetwork
+    - ref: azure-bastion-pre
+    - chain: ipi-install
+    post:
+    - chain: gather-network
+    - chain: gather-core-dump
+    - chain: ipi-deprovision
+    - ref: ipi-deprovision-azure-resourcegroup
+    env:
+      CREATE_BASTION: "yes"
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/private/cucushift-installer-rehearse-azure-ipi-private-workflow.yaml:14]: azure-bastion-pre ref is already added in this workflow, I do not think the `CREATE_BASTION` is needed.

**Pull request closed without merging**